specification/ai/OpenAI.Assistants/tools/tool_resources.tsp

```diff
@@ -0,0 +1,174 @@
+model ToolResources {
+
+  /** Resources to be used by the `code_interpreter tool` consisting of file IDs. */
+  @encodedName("application/json", "code_interpreter")
+  codeInterpreter?: CodeInterpreterToolResource;
+
+  /** Resources to be used by the `file_search` tool consisting of vector store IDs. */
+  @encodedName("application/json", "file_search")
+  fileSearch?: FileSearchToolResource;
+}
+
+/**
+ * A set of resources that are used by the `code_interpreter` tool.
+ */
+@added(ServiceApiVersions.v2024_05_01_preview)
+model CodeInterpreterToolResource {
+  /** 
+   * A list of file IDs made available to the `code_interpreter` tool. There can be a maximum of 20 files 
+   * associated with the tool. 
+   */
+  @encodedName("application/json", "file_ids")
+  @maxItems(20)
+  fileIds: string[] = [];
+}
+
+/**
+ * A set of resources that are used by the `file_search` tool.
+ */
+@added(ServiceApiVersions.v2024_05_01_preview)
+model FileSearchToolResource {
+    /**
+     * The ID of the vector store attached to this assistant. There can be a maximum of 1 vector
+     * store attached to the assistant.
+     */
+    @maxItems(1)
+    @encodedName("application/json", "vector_store_ids")
+    vectorStoreIds?: string[];
+}
+
+//
+// Creation request objects
+//
+
+/**
+ * Request object. A set of resources that are used by the assistant's tools. The resources are specific to the 
+ * type of tool. For example, the `code_interpreter` tool requires a list of file IDs, while the `file_search` 
+ * tool requires a list of vector store IDs.
+ */
+@added(ServiceApiVersions.v2024_05_01_preview)
+model CreateToolResourcesOptions {
+  
+    /** 
+     * A list of file IDs made available to the `code_interpreter` tool. There can be a maximum of 20 files 
+     * associated with the tool. 
+     */
+    @encodedName("application/json", "code_interpreter")
+    codeInterpreter?: CreateCodeInterpreterToolResourceOptions;
+  
+    /** A list of vector stores or their IDs made available to the `file_search` tool. */
+    @encodedName("application/json", "file_search")
+    fileSearch?: CreateFileSearchToolResourceOptions;
+}
+
+/**
+ * A set of resources that will be used by the `code_interpreter` tool. Request object.
+ */
+@added(ServiceApiVersions.v2024_05_01_preview)
+model CreateCodeInterpreterToolResourceOptions {
+    /** A list of file IDs made available to the `code_interpreter` tool. */
+    @maxItems(20)
+    @encodedName("application/json", "file_ids")
+    fileIds?: string[] = [];
+}
+
+#suppress "@azure-tools/typespec-autorest/union-unsupported" "This union is defined according to the OpenAI API"
+/**
+ * A set of resources that are used by the assistant's tools. The resources are specific to the type of tool. 
+ * For example, the `code_interpreter` tool requires a list of file IDs, while the `file_search` tool requires 
+ * a list of vector store IDs.
+ */
+@added(ServiceApiVersions.v2024_05_01_preview)
+@oneOf
```

Comment:
It seems like `anyOf` is the default. This way we can ensure that we correctly represent this in the yaml file. [Source]](https://typespec.io/docs/getting-started/typespec-for-openapi-dev#polymorphism-using-anyof-and-oneof-oas3)